tests/test_coadds.py

```diff
@@ -0,0 +1,335 @@
+            (geom.Point2D(17, 16), Index2D(x=1, y=1)),  # inner point in upper right
+            (geom.Point2D(15, 8), Index2D(x=1, y=0)),  # inner point in lower right
+            (geom.Point2D(0, 29), Index2D(x=0, y=1)),  # inner point in upper left
+            (geom.Point2D(0, 0), Index2D(x=0, y=0)),  # inner point in lower left
+        )
+
+        schema = lsst.meas.base.tests.TestDataset.makeMinimalSchema()
+
+        single_cell_coadds = []
+        cls.exposures = dict.fromkeys(cls.psf_sigmas.keys())
+
+        for x in range(nx):
+            for y in range(ny):
+                identifiers = CellIdentifiers(
+                    cell=Index2D(x=x, y=y),
+                    skymap=common.identifiers.skymap,
+                    tract=common.identifiers.tract,
+                    patch=common.identifiers.patch,
+                    band=common.identifiers.band,
+                )
+
+                outer_bbox = geom.Box2I(
+                    geom.Point2I(x * cls.inner_size, y * cls.inner_size),
+                    geom.Extent2I(cls.inner_size, cls.inner_size),
+                )
+                outer_bbox.grow(cls.border_size)
+
+                dataset = lsst.meas.base.tests.TestDataset(
+                    patch_outer_bbox, psfSigma=cls.psf_sigmas[identifiers.cell]
+                )
+
+                for (inst_flux, position, shape) in sources:
+                    dataset.addSource(inst_flux, position, shape)
+
+                # Create a spatially varying variance plane.
+                variance = ImageD(
+                    np.random.uniform(
+                        0.8,
+                        1.2,
+                        (
+                            nx * cls.inner_size + 2 * cls.border_size,
+                            nx * cls.inner_size + 2 * cls.border_size,
+                        ),
+                    ),
+                    xy0=outer_bbox.getMin(),
+                )
+                exposure, _ = dataset.realize(variance.getArray() ** 0.5, schema, randomSeed=123456789)
+                cls.exposures[identifiers.cell] = exposure
+                exposure = exposure[outer_bbox]
+                image_plane = OwnedImagePlanes(
+                    image=exposure.image, variance=exposure.variance, mask=exposure.mask
+                )
+
+                single_cell_coadds.append(
+                    SingleCellCoadd(
+                        outer=image_plane,
+                        psf=SingleGaussianPsf(
+                            cls.psf_size, cls.psf_size, cls.psf_sigmas[Index2D(x=x, y=y)]
+                        ).computeKernelImage(outer_bbox.getCenter()),
+                        inner_bbox=geom.Box2I(
+                            geom.Point2I(x * cls.inner_size, y * cls.inner_size),
+                            geom.Extent2I(cls.inner_size, cls.inner_size),
+                        ),
+                        inputs={
+                            None,
+                        },
```

Comment:
This should be a FrozenSet[ObservationIdentifiers]`, but this is not being tested right now. So just supplying `None` works for now.